### PR TITLE
wrapGAppsHook: add basic tests

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Wanted to reproduce https://github.com/NixOS/nixpkgs/issues/85515

cc @worldofpeace

### CODE REVIEW

File: pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+PREFIX = $(out)
```
Comment (Ericson2314) [inline pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile:1]: Other tests are in `pkgs/test`, BTW, though I am sympathetic to putting tests near what is being tested, too.

Comment (jtojnar) [reply pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile:1]: Oh, did not notice that. It might be cleaner to move it there.

**Comment by jtojnar** [comment]
One thing I do not like is that everything is in bash some testing library would be nice.

File: pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile (hunk 1 above)
Comment (FRidh) [reply pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile:1]: Python tests are in `pkgs/development/interpreters/python/`

**Comment by FRidh** [comment]
I used pytest for the Python tests but I doubt you would gain much considering the site of the test.

**Comment by jtojnar** [comment]
I verified that #86416 with the following patch successfully passes `typelib-self-user-has-gi-typelib-path` test.


```patch
--- a/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix
+++ b/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix
@@ -13,13 +13,14 @@
 
 makeSetupHook {
   deps = lib.optional (!stdenv.isDarwin) dconf.lib ++ [
-    gtk3
-    librsvg
+    # gtk3
+    # librsvg
     makeWrapper
   ];
   substitutions = {
     passthru.tests = let
       sample-project = ./tests/sample-project;
+      gobject-introspection-setup-hook = makeSetupHook {} gobject-introspection.setupHook;
       skip = cond: lib.optionalString cond ''touch $out; echo "Skipping test $name"; exit'';
     in rec {
       basic = stdenv.mkDerivation {
@@ -58,7 +59,7 @@ makeSetupHook {
         src = sample-project;
 
         nativeBuildInputs = [
-          gobject-introspection
+          gobject-introspection-setup-hook
           wrapGAppsHook
         ];
 
@@ -84,7 +85,7 @@ makeSetupHook {
         src = sample-project;
 
         nativeBuildInputs = [
-          gobject-introspection
+          gobject-introspection-setup-hook
           wrapGAppsHook
         ];
 

```

File: pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile (hunk 1 above)
Comment (jtojnar) [reply pkgs/build-support/setup-hooks/wrap-gapps-hook/tests/sample-project/Makefile:1]: Thinking about it more, I prefer having it near as well.

**Comment by jtojnar** [comment]
I would like to merge this as is as I keep discovering new and new edge cases around the setup hooks.

File: pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    # librsvg provides a module for gdk-pixbuf to allow rendering
+    # SVG icons. Most icon themes are SVG-based and so are some
+    # graphics in GTK (e.g. cross for closing window in window title bar)
+    # so it is pretty much required for applications using GTK.
+    librsvg
+
+    # We use the wrapProgram function.
+    makeWrapper
+  ];
+  substitutions = {
```
Comment (jtojnar) [inline pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix:37]: The `substitutions` make this a mass rebuild due to

https://github.com/NixOS/nixpkgs/blob/c61b6b1b18f9887243f9de4305c412837293738d/pkgs/build-support/trivial-builders.nix#L382-L384

**Pull request merged**